cmd2/cmd2.py

```diff
@@ -448,6 +436,37 @@ def __init__(self, completekey: str='tab', stdin=None, stdout=None, persistent_h
         # If this string is non-empty, then this warning message will print if a broken pipe error occurs while printing
         self.broken_pipe_warning = ''
 
+        # Check if history should persist
+        if persistent_history_file and rl_type != RlType.NONE:
+            persistent_history_file = os.path.expanduser(persistent_history_file)
+            read_err = False
+
+            try:
+                # First try to read any existing history file
+                readline.read_history_file(persistent_history_file)
+            except FileNotFoundError:
+                pass
+            except OSError as ex:
```

Comment:
This is a good change since various exceptions got merged into OSError in Python 3.3 and we no longer support earlier versions of Python